### DISCUSSION TITLE
What are the advantages of using Presto on single data source?

### DISCUSSION THREAD

**Discussion by matanper** [prestodb/presto#15445] [Q&A] [answered]
If I have only a single data source (like Kudu, Mongodb, Druid, Cassandra, etc...) are there any advantages to query from it using presto instead of directly?

**Comment by bitsondatadev** [comment, marked as answer]
@matanper I actually [answered a question about this](https://www.youtube.com/watch?v=TIhNXxkXjls&t=3744s) on the Presto Community Broadcast but the question was geared more towards RDBMSes. The general answer to this is there is very little benefit to adding Presto over a single data source unless you are replacing the Hive runtime with a Presto distribution ([see more on hive connector](https://trino.io/blog/2020/10/20/intro-to-hive-connector.html)). Outside of this use case that Presto was built around, you generally won't get any extra performance benefits of adding a distributed query engine over another distributed database. These databases typically have better optimizations (indexes and cacheing) that make them better choices when only dealing with that data source. To answer the specific questions, some advantages to adding Presto over a single data source would be:

1. ANSI SQL standard over a database that has a complex Query DSL, proprietary SQL flavor, or other interface.
2. You're replacing or augmenting Spark, Impala, Hive to support interactive queries over big data using the hive connector. (Mentioned above)
3. You have the capability to run federated analytics queries over multiple data sources in the future.

This was answered for the Trino distribution (which is a different flavor of Presto formerly known as prestosql) but generally applies to all Presto distributions. Here are the [show notes](https://trino.io/episodes/7.html) for the episode.
